app/models/hyrax/flexible_schema.rb

```diff
@@ -0,0 +1,40 @@
+class Hyrax::FlexibleSchema < ApplicationRecord
+  serialize :profile, coder: YAML
+
+  def attributes_for(class_name)
+    class_names[class_name]
+  end
+
+  def class_names
+    return @class_names if @class_names
+    @class_names = {}
+    profile['classes'].keys.each do |class_name|
+      @class_names[class_name] = {}
+    end
+    profile['properties'].each do |key, values|
+      values['available_on']['class'].each do |property_class|
+        # map some m3 items to what Hyrax expects
+        values = values_map(values)
+        @class_names[property_class][key] = values
+      end
+    end
+    @class_names
+  end
+
+  def values_map(values)
```

Comment:
consider making values_map and lookup_type private

Reply:
I'm fine with this